karabo_data/validation.py

```diff
@@ -160,6 +162,22 @@ def check_files(self):
             fv = FileValidator(f)
             self.problems.extend(fv.run_checks())
 
+    def check_files_map(self):
```

Comment:
Good question. My thinking is that if there's anything weird about accessing data, `karabo-data-validate` will try to find possible causes - including incorrect cache files. Also, if we can get the cache created automatically when the files are migrated/calibrated, it effectively becomes part of the data format.

Maybe we can have a single tool, but with an extra option, e.g. `--check-cache`, to also include checking the cache file.

Reply:
I see, it's probably correct to check them as part of the validation then, fine with me to leave it as it is.

Can't you get the file descriptor with: `file.id.get_vfd_handle()`?

Regarding the cache files 'editing', maybe we could change the file mode and set them as read-only by default, I believe that `karabo_data` will never have to edit them either, right?

Reply:
Oh nice, I had missed that. If that works, we should be good - HDF5 locks files while it writes them, so if we can open a file it shouldn't be changing underneath us, and if we use the fd it can't be replaced before we `stat` it.

Reply:
Seems a bit like overkill, but if people manually editing the cache file is a concern we could save a hash of the file along with it and check that the file matches the hash before loading it. I agree it's not necessary though, don't think anybody would try to edit them.

Reply:
That's true. I keep thinking about this like a security question, i.e. if someone can edit the file, they can also regenerate the hash. But of course we're thinking about unintentionally breaking things.

In any case, I think it's pretty unlikely that people would edit these files. And if we can get them automatically generated in the run folders, users won't have access to modify them anyway.